File: PokerProtocol/src/protocol/commands/game/PlayerPlayMoneyCommand.java
package protocol.commands.game;

import java.util.StringTokenizer;

import protocol.commands.Command;
import protocol.commands.ICommand;

public class PlayerPlayMoneyCommand implements ICommand
{
    
    private final int m_played;
    public static String COMMAND_NAME = "gamePLAY_MONEY";
    
    public PlayerPlayMoneyCommand(StringTokenizer argsToken)
    {
        m_played = Integer.parseInt(argsToken.nextToken());
    }
    
    public PlayerPlayMoneyCommand(int played)
    {
        m_played = played;
    }
    
    @Override
    public String encodeCommand()
    {
        final StringBuilder sb = new StringBuilder();
        sb.append(PlayerPlayMoneyCommand.COMMAND_NAME);
        sb.append(Command.DELIMITER);
        sb.append(m_played);
        return sb.toString();
    }
    
    public int getPlayed()
    {
        return m_played;
    }
}

File: trunk/C#/PokerProtocol/Commands/Game/PlayerPlayMoneyCommand.cs
using System.Text;
using EricUtility;
using EricUtility.Networking.Commands;

namespace PokerProtocol.Commands.Game
{
    public class PlayerPlayMoneyCommand : AbstractTextCommand
    {
        public static string COMMAND_NAME = "gamePLAY_MONEY";

        private readonly int m_Played;

        public int Played
        {
            get { return m_Played; }
        }

        public PlayerPlayMoneyCommand(StringTokenizer argsToken)
        {
            m_Played = int.Parse(argsToken.NextToken());
        }

        public PlayerPlayMoneyCommand(int played)
        {
            m_Played = played;
        }

        public override void Encode(StringBuilder sb)
        {
            Append(sb, m_Played);
        }
    }
}
